WordPress/gutenberg
# Fix/54727 footnote return link accessibility

## What?
Adds aria-labels to the footnote links to make the context more clear for users using accessibility tools.

## Why?
Fixes #54727. During an accessibility audit we were made aware that the footnotes were not accessible enough. The aria-labels offers a solution.

## How?
This pull request simply adds an aria-label to the footnote link and updates the translatable string for the footnote return link to add more clarity.

## Testing Instructions
1. Create a post or page.
2. Insert a few paragraphs and add a footnote. Add a second footnote just for good measure.
3. Save the post or page.
4. Preview the page and check the footnote links for the aria-labels.
